store/src/main/java/org/apache/rocketmq/store/DefaultMessageStore.java

```diff
@@ -510,17 +511,20 @@ private boolean isLmqConsumeQueueNumExceeded() {
     @Override
     public PutMessageResult putMessage(MessageExtBrokerInner msg) {
         try {
-            return asyncPutMessage(msg).get();
-        } catch (InterruptedException | ExecutionException e) {
+            // to make sure put timeout bigger than flush timeout, full gc is also considered
+            int putTimeout  = this.messageStoreConfig.getSyncFlushTimeout() + 5000;
```

Comment:
IMO, it would be better to add a new messageStoreConfig like putMessageTimeout than using syncFlushTimeout + 5000.